flowmachine/flowmachine/core/cache.py

```diff
@@ -0,0 +1,384 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# -*- coding: utf-8 -*-
+
+"""
+Functions which deal with inspecting and managing the query cache.
+"""
+import logging
+import pickle
+
+from typing import TYPE_CHECKING, Tuple, List
+
+from psycopg2 import InternalError
+
+if TYPE_CHECKING:
+    from .query import Query
+    from .connection import Connection
+
+logger = logging.getLogger("flowmachine").getChild(__name__)
+
+
+def touch_cache(connection: "Connection", query_id: str) -> float:
+    """
+    'Touch' a cache record and update the cache score for it.
+
+    Parameters
+    ----------
+    connection : Connection
+    query_id : str
+        md5 id of the query to touch
+
+    Returns
+    -------
+    float
+        The new cache score
+    """
+    try:
+        return float(connection.fetch(f"SELECT touch_cache('{query_id}')")[0][0])
+    except (IndexError, InternalError):
+        raise ValueError(f"Query id '{query_id}' is not in cache on this connection.")
+
+
+def reset_cache(connection: "Connection") -> None:
+    """
+    Reset the query cache. Deletes any tables under cache schema, resets the cache count
+    and clears the cached and dependencies tables.
+
+    Parameters
+    ----------
+    connection : Connection
+    """
+    qry = f"SELECT tablename FROM cache.cached WHERE NOT class='Table'"
```

Comment:
I _think_ the logic here is correct, but it's a bit confusing that the docstring says it deletes any tables under the cache schema and then you select all items where `class` is _not_ `'Table'`. Is there any way of clarifying things a bit more? (Not sure what to suggest because I'm a bit hazy on exactly what lives in the cache schema.

Reply:
I see, thanks for the explanation. Related to my comment a bit further down, do you think in the medium run it is worth storing Table queries elsewhere so that we don't need to treat them as exceptions all over the place?

Reply:
Definitely worth investigating that.